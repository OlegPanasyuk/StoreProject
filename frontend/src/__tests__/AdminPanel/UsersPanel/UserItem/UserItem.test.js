import React from 'react';
import UserItemConnected, { UserItem } from '../../../../AdminPanel/UsersPanel/UserItem';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    obj: {
        id: 1,
        username: 'User',
        role: 'Admin'
    },
    editUser: ()=>{}
}) {
    let component = shallow(<UserItem {...props} />);
    return { component, props };
}

describe('UserItem --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(UserItem.propTypes.obj).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});


describe('UserItem ', () => {
    let initialState = {
        
    };

    const mockStore = configureStore();
    let store, wrapper;
    let props = {
        obj: {
            id: 1,
            username: 'User',
            role: 'Admin'
        },
        editUser: ()=>{}
    };
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><UserItemConnected {...props}/></Provider>);
    });
    it('+++ render the connected(SMART) component',() => {
        expect(wrapper.find(UserItemConnected).length).toEqual(1);
    });
});

