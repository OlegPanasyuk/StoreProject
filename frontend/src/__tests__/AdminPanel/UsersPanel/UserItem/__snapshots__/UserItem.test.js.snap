// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem --- Snapshot Component must be rendered 1`] = `
<Card
  as="div"
  body={false}
>
  <CardHeader>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      <i
        className="far fa-edit"
      />
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      <i
        className="fas fa-trash-alt"
      />
    </Button>
  </CardHeader>
  <CardBody>
    <CardTitle>
      User
    </CardTitle>
    <CardSubtitle
      className="mb-2 text-muted"
    >
      Admin
    </CardSubtitle>
  </CardBody>
</Card>
`;
